### PR TITLE
icon-layer-triangle-issue

### PR DESCRIPTION
**Pull request by ellie-farjoud**
Replaced TRIANGLE_FAN with TRIANGLES, since IE11 is not supporting TRIANGLE_FAN

### CODE REVIEW

**Comment by ibgreen** [comment]
There are quite a few layers that use `TRIANGLE_FAN` @tsherif

**Comment by ibgreen** [comment]
@tsherif I wonder if we could handle this on a luma/polyfill level - i.e. if we get a `TRIANGLE_FAN` on IE11 we generated an expanded buffer and replace with `TRIANGLE` under the hood.

That way we could keep the deck.gl code clean...

File: modules/layers/src/icon-layer/icon-layer.js [hunk 1]
```diff
@@ -200,14 +200,14 @@ export default class IconLayer extends Layer {
   }
 
   _getModel(gl) {
-    const positions = [-1, -1, 0, -1, 1, 0, 1, 1, 0, 1, -1, 0];
+    const positions = [-1, -1, 1, -1, 1, 0, 1, 1, 0, 1, -1, 1, -1, -1, 1, -1, -1, 1];
```
Comment (Pessimistress) [inline modules/layers/src/icon-layer/icon-layer.js]: z values should be `0`.

Comment (ellie-farjoud) [reply modules/layers/src/icon-layer/icon-layer.js]: It 's fixed.

**Comment by tsherif** [comment]
TRIANGLE_FAN works fine in IE. Can you share a bit about why you think otherwise? Are you getting errors with deck.gl?

**Comment by ellie-farjoud** [comment]
Actually the TRIANGLE_FAN doesn't work fine in ie11, while TRIANGLES does work fine in IE11.

I do not get any error in deck.gl. 

Here is the issue I had: https://github.com/uber/deck.gl/issues/2486

and by changing the positions, found that the triangle-fan is not supported in IE11.

By converting The TRIANGLE_FAN to TRIANGLES the icons is shown completely as below:

fixed-iconlayer

**Comment by ellie-farjoud** [comment]
I do not get any error in deck.gl.

Here is the issue I had: #2486

and by changing the positions from TRIANGLE_FAN to TRIANGLE the icons are shown completely as below:

fixed-iconlayer

If IE11 does support the TRIANGLE_FAN, still something happening that The TRIANGLE_FAN is converted to TRIANGLES.

**Comment by tsherif** [comment]
Can you try loading this page in IE? https://tsherif.github.io/triangle-fan-test/gl.html

This is what it should look like:
Screen Shot 2019-09-19 at 5 42 42 PM

**Comment by ellie-farjoud** [comment]
Yes, it looks same.

**Comment by tsherif** [comment]
That scene is using TRIANGLE_FAN so that's not the root issue. We made some fixes recently in luma.gl to how drawing is handled in IE (https://github.com/uber/luma.gl/pull/1238). I'm curious if this is related. Could you try your code against a beta version of deck.gl? I believe the latest is 7.3.0-beta.2.

**Comment by ellie-farjoud** [comment]
Yes, it's working, thanks so much!

**Comment by tsherif** [comment]
Great!

**Comment by etimberg** [comment]
@tsherif @ibgreen I tested the luma.gl change I made, but it doesn't resolve this issue. The icon layer still draws incorrectly. It almost looks like some kind of transform has been applied to the icon square turning it into a rotated rectangle.

bad icons

**Comment by etimberg** [comment]
@tsherif @ibgreen I've done some debugging on this and I think I have a cause, but I'm not yet sure what the ultimate root cause is.

IE11 appears to be interpreting the list of positions as a set of `x,y` points rather than as a set of `x,y,z` points. If I override `IconLayer._getModel()` to change the positions to be 2d instead of 3d, IE11 works fine and Chrome breaks.

Any ideas on what might be causing this?

**Pull request closed without merging**